File: test/csharp/SpecGate.CtscFixtures/Conformance/Basic/CheckpointInline.cs
using SpecGate.Annotations;

namespace SpecGate.CtscFixtures.Conformance.Basic;

public static class CheckpointInline
{
    [SpecOperation("after_upper", Spec = "fixture.checkpoint_inline")]
    public static string AfterUpper([SpecInput("value")] string value) => value;

    [SpecOperation("process", Spec = "fixture.checkpoint_inline")]
    public static string Process([SpecInput("data")] string data)
    {
        string upper = data.ToUpperInvariant();
        _ = AfterUpper(upper);
        return upper.Trim();
    }
}

File: test/rust/crates/specgate-ctsc-fixtures/src/conformance/basic/checkpoint_inline.rs
use specgate::spec_operation;

#[spec_operation("after_upper", spec = "fixture.checkpoint_inline")]
pub fn after_upper(value: &str) -> String {
    value.to_string()
}

#[spec_operation("process", spec = "fixture.checkpoint_inline")]
pub fn process(data: &str) -> String {
    let upper = data.to_uppercase();
    after_upper(&upper);
    upper.trim().to_string()
}

#[test]
fn records_the_intermediate_uppercase_observation() {
    assert_eq!(process("  hello  "), "HELLO");
}
